### PR TITLE
Fix deprovision of avs evaluations for trial/freemium plan

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

If the `De-provision_AVS_Evaluations` step was run again for the trial/freemium plan, even if the internal evaluation was removed successfully in the previous run, the step was not skipped.

Changes proposed in this pull request:

- skip `De-provision_AVS_Evaluations` if internal evaluation have been deleted for trial/freemium plan

**Related issue(s)**

See also #2518

### CODE REVIEW

File: components/kyma-environment-broker/internal/process/deprovisioning/avs_evaluations.go [hunk 1]
```diff
@@ -37,9 +37,15 @@ func (ars *AvsEvaluationRemovalStep) Name() string {
 
 func (ars *AvsEvaluationRemovalStep) Run(operation internal.Operation, logger logrus.FieldLogger) (internal.Operation, time.Duration, error) {
 	logger.Infof("Avs lifecycle %+v", operation.Avs)
-	if operation.Avs.AVSExternalEvaluationDeleted && operation.Avs.AVSInternalEvaluationDeleted {
-		logger.Infof("Both internal and external evaluations have been deleted")
-		return operation, 0, nil
+	if operation.Avs.AVSInternalEvaluationDeleted {
```
Comment (jaroslaw-pieszka) [inline components/kyma-environment-broker/internal/process/deprovisioning/avs_evaluations.go:40]: How about two functions:

1. logic for trial or freemium
2. logic for remaining plans

 Wouldn't it be more readable and bring us closer to extraction on step level (possibly in the future).

Comment (jaroslaw-pieszka) [reply components/kyma-environment-broker/internal/process/deprovisioning/avs_evaluations.go:40]: Or much simpler.
Each call of DeleteAVSEvaluation is guarded with flag checking.
We wouldn't increase the complexity, nesting level and we would avoid code duplication which is present in your solution.

File: resources/kcp/charts/kyma-environment-broker/values.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ global:
   images:
     kyma_environment_broker:
       dir:
-      version: "PR-2513"
+      version: "PR-2519"
```
Comment (jaroslaw-pieszka) [inline resources/kcp/charts/kyma-environment-broker/values.yaml:5]: No test cases added.

**Comment by MarekMichali** [comment]
Upon further investigation, the original logic is correct, but requires some refactoring to clean up redundant code that is confusing. A separate ticket will be opened.

**Pull request closed without merging**